processing-service-core/api/models/db/repositoryConfigStore.js

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright 2017-2018 Micro Focus or one of its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+// handles communication requests with tenant config database table
+const Sequelize = require('sequelize');
+const Q = require('q');
+const apiErrorFactory = require('../errors/apiErrorFactory.js');
+const databaseDefinition = require('./databaseConnection.js').definition;
+const logger = require('../../helpers/loggingHelper.js');
+const repositoryConfigTableDetails = require('./tables/repositoryConfig.js');
+
+module.exports = {
+    deleteRepositoryConfig: deleteRepositoryConfig,
+    getRepositoryConfig: getRepositoryConfig,
+    getRepositoryConfigs: getRepositoryConfigs,
+    setRepositoryConfig: setRepositoryConfig
+}
+
+const tableDefinition = repositoryConfigTableDetails.definition;
+const repositoryConfigTable = databaseDefinition.define(repositoryConfigTableDetails.name,
+    tableDefinition,
+    { tableName:  repositoryConfigTableDetails.tableName, timestamps: false });
+
+/**
+ * Deletes a repository config matching the specified parameters.
+ * @param repositoryId {String} the repository ID identifying the config that should be deleted.
+ * @param tenanatId {String} the tenant ID identifying the config that should be deleted.
+ * @param key {String} the key identifying the config that should be deleted.
+ * @returns {*|d.promise|Function|promise|a|h} a promise that will be resolved or rejected based on result of delete.
+ *  Resolved promise will return an empty object.
+ *  Rejected promise will pass an ApiError with either type set to ITEM_NOT_FOUND if no repository config with the specified repository ID
+ *  and key could be found or type DATABASE_UNKNOWN_ERROR for any other kind of failure.
+ */
+function deleteRepositoryConfig(repositoryId, tenantId, key) {
```

Comment:
I know it's minor but I feel that `tenantId` should come first in all of these methods; before `repositoryId`.